pkg/controllers/monitoring/monitoring-stack/conditions.go

```diff
@@ -0,0 +1,121 @@
+package monitoringstack
+
+import (
+	monv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	stack "github.com/rhobs/observability-operator/pkg/apis/monitoring/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	monitoringStackAvailable       = "MonitoringStackAvailable"
+	monitoringStackReconciled      = "MonitoringStackReconciled"
+	prometheusNotAvailable         = "PrometheusNotAvailable"
+	prometheusNotReconciled        = "PrometheusNotReconciled"
+	cannotReadPrometheusConditions = "Cannot read Prometheus status conditions"
+)
+
+type conditions map[stack.ConditionType]*stack.Condition
+
+type conditionsMap struct {
+	conditions
+}
+
+// CreateConditions creates a new map with MonitoringStack status conditions.
+// The newly created conditions are in unknown state.
+func CreateConditions() conditionsMap {
+	cm := make(map[stack.ConditionType]*stack.Condition)
+	cm[stack.Available] = &stack.Condition{
+		Type:               stack.Available,
+		Status:             stack.ConditionUnknown,
+		LastTransitionTime: metav1.Now(),
+	}
+
+	cm[stack.Reconciled] = &stack.Condition{
+		Type:               stack.Reconciled,
+		Status:             stack.ConditionUnknown,
+		LastTransitionTime: metav1.Now(),
+	}
+	return conditionsMap{conditions: cm}
+}
+
+// UpdateAvailable gets existing "Available" condition and updates its parameters
+// based on the Prometheus "Available" condition
+func (cm *conditionsMap) UpdateAvailable(prometheusConditions []monv1.PrometheusCondition, generation int64) {
+	ac := cm.conditions[stack.Available]
+	ac.ObservedGeneration = generation
+
+	prometheusAvailable := getPrometheusCondition(prometheusConditions, monv1.PrometheusAvailable)
+
+	if prometheusAvailable == nil {
+		ac.Status = stack.ConditionFalse
+		ac.Reason = prometheusNotAvailable
+		ac.Message = cannotReadPrometheusConditions
+		ac.LastTransitionTime = metav1.Now()
+		return
+	}
+
+	if prometheusAvailable.Status != monv1.PrometheusConditionTrue {
+		ac.Status = stack.ConditionFalse
+		ac.Reason = prometheusNotAvailable
+		ac.Message = prometheusAvailable.Message
+		ac.LastTransitionTime = metav1.Now()
```

Comment:
Ideally we should only update the transition time if the condition has changed?
Also we don't update the transition time if the status is true.